src/workspaces/list/WorkspacesListTabs.ts

```diff
@@ -64,14 +64,25 @@ export const WorkspacesListTabs = (props: WorkspacesListTabsProps): ReactNode =>
   );
 };
 
-const filterWorkspaces = (workspaces: CategorizedWorkspaces, filters: WorkspaceFilterValues): CategorizedWorkspaces => {
+// exported for unit testing
+export const filterWorkspaces = (
+  workspaces: CategorizedWorkspaces,
+  filters: WorkspaceFilterValues
+): CategorizedWorkspaces => {
   const filterWorkspacesCategory = (workspaces: Workspace[], filters: WorkspaceFilterValues): Workspace[] => {
     const matches = (ws: Workspace): boolean => {
       const {
-        workspace: { namespace, name, attributes },
+        workspace: { namespace, name, attributes, state, googleProject },
       } = ws;
+
+      const containsKeywordFilter = (textToMatchWithin: string | undefined): boolean => {
+        return textToMatchWithin === undefined ? false : textMatch(filters.keywordFilter, textToMatchWithin!);
```

Comment:
`textMatch` does not accept undefined, but `state` and `googleProject` may be undefined.

Reply:
Would it make sense to have textMatch be resilient to undefined arguments?